### PR TITLE
feat(quantic): Update the styling of the rephrase buttons

### PR DESCRIPTION
**Pull request by erocheleau**
SFINT-5426

Before:
image

After:
image

Focus indicators for a11y:
image

VoiceOver, when entering the options: 
image

VoiceOver, announcing the selected state of the option:
image

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+   */
+  @api
+  get iconSize() {
+    return this._iconSize;
+  }
+  set iconSize(value) {
+    if (['xx-small', 'x-small', 'small', 'medium', 'large'].includes(value)) {
+      this._iconSize = value;
+    }
+  }
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:56]: Why does the iconSize needs a getter and a setter? from me it can be just a simple property.

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  @track _value;
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:58]: I don't think the @track is necessary here.

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ * The `QuanticRadioButtonsGroupOption` type defines the shape of an option
+ * that can be passed to the `QuanticRadioButtonsGroup` component.
+ * @typedef {Object} QuanticRadioButtonsGroupOption
+ * @property {String} value The value of the option.
+ * @property {String} [label] The label of the option.
+ * @property {String} [iconName] The name of the icon to be displayed.
+ * @property {String} [tooltip] The tooltip of the option.
+ */
+
+export default class QuanticRadioButtonsGroup extends LightningElement {
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:13]: Please add a documentation for this new component,
example: 
```suggestion
/**
 * The `QuanticRadioButtonGroup` displays ...
 * @category Internal
 * @example
 * <c-quantic-radio-buttons-group options...>
 * </c-quantic-radio-buttons-group>
 */
export default class QuanticRadioButtonsGroup extends LightningElement {
```

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+const selectors = {
+  legend: 'legend.radio-buttons-group__legend',
+  radioInputs: 'input[type="radio"]',
+  radioLabels: 'label.radio-buttons-group__label',
+  radioIcons: 'lightning-icon',
+  radioContainers: '.slds-button.slds-radio_button.radio-button',
+  radioTooltips: 'c-quantic-tooltip',
+};
+
+const defaultOptionsList = [
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:14]: This will allow you to well define the typings and avoid adding ts-ignors in the code.
```suggestion
/**
 * @type {Array<{value: string, tooltip?: string, label?: string }>}
 */
const defaultOptionsList = [
```

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+  });
+
+  describe('default rendering options', () => {
+    it('should render correctly with default options, without labels, without icons, without tooltips', async () => {
+      const minimalOptions = [
+        {value: 'value_option1'},
+        {value: 'value_option2'},
+        {value: 'value_option3'},
+      ];
+      // @ts-ignore
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:76]: I dont' think this necessary.

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+      const element = createTestComponent({
+        ...exampleOptions,
+        // @ts-ignore
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:79]: no longer necessary.
```suggestion
```

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+      expect(radioInputs).toHaveLength(minimalOptions.length);
+      expect(radioLabels).toHaveLength(0);
+      expect(icons).toHaveLength(0);
+      minimalOptions.forEach((option, index) => {
+        expect(radioInputs[index].value).toBe(option.value);
+      });
+    });
+
+    it('should render correctly with default options, without icons', async () => {
+      // @ts-ignore
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:104]: Why do we need a ts-ignore here?

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+      expect(
+        radioButtonContainers[0].classList.contains('radio-button--selected')
+      ).toBeFalsy();
+      expect(
+        radioButtonContainers[1].classList.contains('radio-button--selected')
+      ).toBeTruthy();
+      expect(selectedRadioInputs).toHaveLength(1);
+    });
+
+    it('should dispatch the "focus" event when a radio button is focused', async () => {
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:273]: Is it possible to have the same test, but this time for the blur event?

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+  /** @type {'xx-small' | 'x-small' | 'small' | 'medium' | 'large'} */
+  _iconSize = 'xx-small';
+
+  get radioButtonElements() {
+    return this.template.querySelectorAll('input');
+  }
+
+  get transformedOptions() {
+    const {options, value} = this;
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:68]: I suggest to rename like the following to avoid confusions between `value` and `option.value`.
```suggestion
    const {options, value: selectedValue} = this;
```

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    if (Array.isArray(options)) {
+      return options.map((option, index) => ({
+        label: option.label,
+        value: option.value,
+        iconName: option.iconName,
+        iconVariant: value === option.value ? 'inverse' : '',
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:74]: After renaming, we would need to update the occurrences of `value` to be like the following:
```suggestion
        iconVariant: selectedValue === option.value ? 'inverse' : '',
```

**Review by mmitiche — COMMENTED** [review]
Very nice work! well done 👍

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 11]
```diff
@@ -315,26 +319,6 @@ describe('quantic-generated-answer', () => {
...
-                Expect.searchQueryContainsCorrectRephraseOption(
-                  defaultRephraseOption
-                );
-                if (analyticsMode === 'legacy') {
-                  Expect.logRephraseGeneratedAnswer(
-                    defaultRephraseOption,
-                    thirdStreamId
-                  );
-                }
-              });
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:337]: We can no longer "unselect" the rephrase buttons. The default one is now "auto" and it's selected by default.
That's part of the re-design, it was confusing to have 3 buttons as options but be able to unselect them to access a 4th option.
Now it's just 4 options always visible.

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 12]
```diff
@@ -1337,6 +1337,20 @@
...
+    <protected>false</protected>
+    <shortDescription>Auto rephrase</shortDescription>
+  </labels>
+  <labels>
+    <fullName>quantic_Automatic</fullName>
+    <value>Automatic</value>
+    <language>en_US</language>
+    <protected>false</protected>
+    <shortDescription>Automatic rephrase tooltip</shortDescription>
+  </labels>
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:1353]: Well the label on the button is "Auto" while the tooltip is "Automatic"?

The label must be shorter imo to take less space

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerRephraseButtons/quanticGeneratedAnswerRephraseButtons.html [hunk 13]
```diff
@@ -1,24 +1,33 @@
...
         key={option.label}
+        without-borders
         selected={option.isSelected}
         tooltip={option.tooltip}
         label={option.label}
         icon-name={option.iconName}
         onselect={option.handleSelect}
-        ondeselect={handleDeselect}
       ></c-quantic-stateful-button>
     </template>
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerRephraseButtons/quanticGeneratedAnswerRephraseButtons.html:31]: woops, I kept them to make sure I didn't forget to move anything to the new buttons :)

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerRephraseButtons/quanticGeneratedAnswerRephraseButtons.js [hunk 14]
```diff
@@ -61,20 +71,24 @@ export default class QuanticGeneratedAnswerRephraseButtons extends LightningElem
...
-      label: this.getRephraseButtonLabel(option.value),
-      isSelected: this.isSelected(option.value),
-      handleSelect: (event) => {
-        event.stopPropagation();
-        this.handleRephrase(option.value);
-      },
+      value: option.value,
+      label: this.rephraseButtonLabels[option.value],
+      iconName: option.iconName,
+      tooltip: option.tooltip,
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerRephraseButtons/quanticGeneratedAnswerRephraseButtons.js:77]: yea good point, I coded these one by one :)

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerRephraseButtons/quanticGeneratedAnswerRephraseButtons.js [hunk 15]
```diff
@@ -61,20 +71,24 @@ export default class QuanticGeneratedAnswerRephraseButtons extends LightningElem
...
   }
 
   isSelected(option) {
     return this.value === option;
   }
 
+  handleRephraseChange(event) {
+    event.stopPropagation();
+    if (!this.isSelected(event.detail.value)) {
+      this.handleRephrase(event.detail.value);
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerRephraseButtons/quanticGeneratedAnswerRephraseButtons.js:88]: Yes makes sense I changed it to "dispatchRephraseEvent"

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 1 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:56]: I copied it from your stateful button ;) 
Why did you add that getter and setter? 😂

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 2 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:58]: I'm always worried of the "magic" that LWC made to track values haha, but I can remove it.

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 3 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:13]: Even if it's Internal?

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js (hunk 7 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:104]: dunno was probably left during my testing

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js (hunk 8 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:273]: I don't really know how to trigger a blur event on a radio input, tbh I'm not even sure we _can_ blur on a radio button.
I only put it there because I took the same mechanics as the original LWC radio-buttons-group to start with and it was there.

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+        indexId: `radio-${index}`,
+        tooltip: option.tooltip,
+        mouseEnterHandler: () => {
+          const tooltipToShow = this.getTooltipComponentByValue(option.value);
+          tooltipToShow.showTooltip();
+        },
+        mouseLeaveHandler: () => {
+          const tooltipToHide = this.getTooltipComponentByValue(option.value);
+          tooltipToHide.hideTooltip();
+        },
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:87]: Since there are so many properties there, I would much rather have them explicitly listed.

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+    return `slds-button slds-radio_button ${paddingClass} radio-button radio-button--${option.value === this.value ? 'selected' : 'unselected'}`;
+  }
+
+  handleFocus() {
+    this.dispatchEvent(new CustomEvent('focus'));
+  }
+
+  handleBlur() {
+    this.dispatchEvent(new CustomEvent('blur'));
+  }
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:117]: This comes from the base implementation of the radio-buttons-group from the LWC components library.
I think this was mostly there so we could handle "dirty" data, in the sense if those radio buttons were in a form that would have validation after you change a value for example.

They are not really useful here with the current usage of the radio-buttons-group. It's more of a "could be useful later"

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+   */
+  @api legend = '';
+  /**
+   * The list of options to be displayed.
+   * @api
+   * @type {QuanticRadioButtonsGroupOption[]}
+   */
+  @api options;
+  /**
+   * If the labels should be hidden or visible.
```
Comment (fbeaudoincoveo) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:33]:
```suggestion
   * Whether the labels should be hidden.
```

It's a total pet peeve of mine, though, so feel free to ignore 😂

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 18 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:33]: I think mine is a lot easier to read quickly lol :P

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 1 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:62]: Ah, I see, it was to keep the default value of the icon size to the value `xx-small`. 
so in the case where an invalid value other than ['xx-small', 'x-small', 'small', 'medium', 'large'] is provided, we would use the size  `xx-small` instead of the default value that SF usually falls back to which is `medium`

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 3 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:19]: Yeah so we know it is internal x)

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:126]: It's not clear for me how this could be concretely useful  in our use case, so I would suggest to remove it unless you do have a clear idea of how we could benefit from this later in the futur.

Cause I think this is typically the things we add  and their purpose is not really clear and in the future we will be always afraid of removing them fairing to break something.

**Comment by mmitiche** [comment]
I tried the component locally and I noticed the following issues:

- I'm not able to click the rephrase buttons using the mouse, I'm only able to select an option if I use my keyboard to navigate to the buttons:

https://github.com/coveo/ui-kit/assets/86681870/f3751e2b-f886-485e-8427-aabecb009a18

- When the labels of the rephrase buttons are displayed, the buttons are not displayed correctly:
Screenshot 2024-04-09 at 9 38 25 AM

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+                type="radio"
+                class="radio-input--no-outline"
+                name={option.label}
+                id={option.indexId}
+                checked={option.isChecked}
+                value={option.value}
+                onchange={handleChange}
+                onfocus={handleFocus}
+                onblur={handleBlur}
+              />
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html:40]: I believe the reason you have 1 of 1 in the screen reader instead of 1 of 4 is that the radio buttons are grouped together using the same name. If they have different names (here option.label) then html thinks that they are not related if I recall properly. 

https://www.w3.org/WAI/tutorials/forms/grouping/#radio-buttons

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:126]: yea good point 👍

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html (hunk 19 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html:40]: Ohhh perhaps? Lemme test that.
I couldn't figure out the difference between the generated html from my component and the generated html from the salesforce radio-button-group component. Perhaps this is it

Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html:40]: eyyyy it works nice!
image

**Comment by mmitiche** [comment]
From the mockups:https://www.figma.com/file/l63i3ZXWrVxldOR4WUBKS1/RGA-Component?type=design&node-id=3885-13905&mode=design&t=4VoHetYxb2v35HWx-0

Screenshot 2024-04-09 at 5 16 14 PM

when the rephrase buttons are displayed in a new line, they must take the whole width, that's for example how they will need to be displayed in the Hosted Insight. Panel.

Was this considered? can we apply this?

**Review by mmitiche — APPROVED** [review]
Looks good to me, just one concern mentioned related to the difference between the actual implementation and the mockups.

**Review by SimonMilord — APPROVED** [review]
Few minor comments, but looks good!

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js [hunk 20]
```diff
@@ -0,0 +1,289 @@
...
+
+      expect(radioLabels).toHaveLength(defaultOptionsList.length);
+      for (const radioLabel of radioLabels) {
+        expect(radioLabel.classList.contains(labelHiddenClass)).toBeFalsy();
+      }
+    });
+  });
+
+  describe('User interaction', () => {
+    it('should dispatch the "change" event when a radio button is clicked', async () => {
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js:202]: I don't think there's a convention

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html [hunk 21]
```diff
@@ -0,0 +1,57 @@
+<!-- sldsValidatorIgnore -->
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.html:1]: It's to remove all the red lines that the slds validator creates around mobile stuff
image

I do have a label for all the clickable stuff but it still highlights the entire file in yellow xD

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 22]
```diff
@@ -1337,6 +1337,20 @@
...
+    <language>en_US</language>
+    <protected>false</protected>
+    <shortDescription>Auto rephrase</shortDescription>
+  </labels>
+  <labels>
+    <fullName>quantic_Automatic</fullName>
+    <value>Automatic</value>
+    <language>en_US</language>
+    <protected>false</protected>
+    <shortDescription>Automatic rephrase tooltip</shortDescription>
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:1352]: Yup I had forgotten, I added them this am
